src/modules/dashboard/components/ActionsPageFeed/ActionsPageFeedItem/ActionsPageFeedItem.tsx

```diff
@@ -33,6 +35,33 @@ const ActionsPageFeedItem = ({
       <UserMention username={usernameWithAtSign.slice(1)} />
     ),
   });
+
+  const { data: annotationJSON } = useDataFetcher(
+    ipfsDataFetcher,
+    [annotation ? (comment as string) : ''], // Technically a bug, shouldn't need type override
+    [annotation ? comment : ''],
+  );
+
+  const getAnnotationMessage = useCallback(() => {
+    if (!annotationJSON) {
+      return undefined;
+    }
+    const annotationObject = JSON.parse(annotationJSON);
+    if (annotationObject && annotationObject.annotationMessage) {
+      return annotationObject.annotationMessage;
+    }
+    return undefined;
```

Comment:
I don't like how that would look :) (from a code cleanliness point of view)

If I took your example, it will end up looking like:
```
   if (annotationJson) {
     const annotationObject = JSON.parse(annotationJSON);
     if (annotationObject && annotationObject.annotationMessage) {
        return annotationObject.annotationMessage;
      }
      return undefined;
   }
   return undefined;
```

And this is because I explicitly want to return false for cases where the annotation json object isn't "proper", and doing as you suggested will make the code harder to understand.

This is because we have a lot of edge cases here, as this value can be added manually, directly on chain, without our control.

So I prefer to leave it as I originally implemented it.

Reply:
Wouldn't this method return undefined either way, if you remove both "return undefined", and it won't fulfill ifs?